File lwjgl-2.7.1.orig/.ipc_in is a fifo while file lwjgl-2.7.1/.ipc_in is a fifo
File lwjgl-2.7.1.orig/.ipc_out is a fifo while file lwjgl-2.7.1/.ipc_out is a fifo
diff -U 2 -pr lwjgl-2.7.1.orig/work/src/java/org/lwjgl/LinuxSysImplementation.java lwjgl-2.7.1/work/src/java/org/lwjgl/LinuxSysImplementation.java
--- lwjgl-2.7.1.orig/work/src/java/org/lwjgl/LinuxSysImplementation.java	2011-02-06 19:23:32.000000000 -0600
+++ lwjgl-2.7.1/work/src/java/org/lwjgl/LinuxSysImplementation.java	2011-11-12 18:41:33.823859307 -0600
@@ -32,4 +32,8 @@
 package org.lwjgl;
 
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedExceptionAction;
+import java.lang.UnsatisfiedLinkError;
 
 /**
@@ -43,5 +47,19 @@ final class LinuxSysImplementation exten
 
 	static {
-		java.awt.Toolkit.getDefaultToolkit(); // This will make sure libjawt.so is loaded
+		// Load libawt.so and libmawt.so, needed for libjawt.so
+		java.awt.Toolkit.getDefaultToolkit();
+		
+		// manually load libjawt.so into vm, needed since Java 7
+		AccessController.doPrivileged(new PrivilegedAction<Object>() {
+			public Object run() {
+				try {
+					System.loadLibrary("jawt");
+				} catch (UnsatisfiedLinkError e) {
+			        // catch and ignore an already loaded in another classloader 
+					// exception, as vm already has it loaded
+			    }
+				return null;
+			}
+		});
 	}
 
